### PR TITLE
streamline npm test

### LINKED ISSUE #266: remove duplication from `npm test` script
Since I forgot to do this in #249, I'm opening this issue for it.

The `npm test` command contains some duplicated steps, specifically in starting Storybook and servers more than once.

Rather than calling other npm scripts in `npm test`, write the test script from scratch. Start the necessary servers, then Storybook, then run the different tests. (Unit tests aren't affected by this and can be called with `npm run test:unit` as usual.)

### PR DESCRIPTION
**Pull request by vadekh**
**Related Issue:**

closes #266

**Description of Changes**

Streamlined the storybook testing scripts so that the storybook server is only started once.
Reordered the five test types in an effort to produce more concise results (minimize extra lines printed by the utilities in between results).

@dwhieb feel free to revert the reordering change if you had a reason for the original order.

### CODE REVIEW

**Comment by dwhieb** [comment]
> @dwhieb feel free to revert the reordering change if you had a reason for the original order.

I was following the typical order of the testing pyramid from shortest > longest on the "fail early, fail fast" principle. It doesn't make that big of a difference, but the idea is that if a unit test fails you know right away, and can ignore the higher-order tests that depend on the unit tests until the unit test is fixed.

I was also just thinking it would be nice if the tests exited at the first failure, so I found the `--bail` flag [here](https://mochajs.org/#command-line-usage) for Mocha, and I guess [Cypress already gives a non-zero exit code if there's a failure](https://docs.cypress.io/guides/guides/command-line#Exit-code)?

So `test:unit` can be updated to `mocha --bail src/**/*.unit.js test/*.unit.js`.

File: package.json [hunk 1]
```diff
@@ -38,7 +38,8 @@
     "start:storybook" : "start-storybook -p 6006 -s dist --ci",
     "stop-only"       : "stop-only -f test,src",
     "storybook"       : "start-storybook -p 6006",
-    "test"            : "npm run test:unit && npm run test:component && npm run test:integration && npm run test:e2e && npm run test:perf",
+    "ct-and-it"       : "npm run cypress-ct-run && npm run cypress-it-run",
```
Comment (dwhieb) [inline package.json]:
```suggestion
    "test:storybook"  : "npm run cypress-ct-run && npm run cypress-it-run",
```

File: package.json [hunk 2]
```diff
@@ -38,7 +38,8 @@
...
+    "test"            : "start-server-and-test start:storybook 6006 ct-and-it && npm run test:e2e && npm run test:unit && npm run test:perf",
```
Comment (dwhieb) [inline package.json]:
```suggestion
    "test"            : "npm run test:unit && start-server-and-test start:storybook 6006 test:storybook && npm run test:e2e && npm run test:perf",
```

**Review by dwhieb — APPROVED** [review]

**Comment by vadekh** [comment]
@dwhieb I added the `--bail` flag, and also tested to ensure that the test run ends after a given section if it fails, and it does not continue to the next type of test.

**Comment by dwhieb** [comment]
Awesome thanks!

**Review by dwhieb — APPROVED** [review]

**Pull request merged**